packages/ember-headless-form/src/components/headless-form.ts

```diff
@@ -37,7 +37,7 @@ export interface HeadlessFormComponentSignature<DATA extends UserData> {
     dataMode?: 'mutable' | 'immutable';
     validateOn?: ValidateOn;
     revalidateOn?: ValidateOn;
-    validate?: FormValidateCallback<FormData<DATA>>;
+    validate?: FormValidateCallback<DATA>;
```

Comment:
Just curious, why did FormData get removed here?

Reply:
Yeah, tbh, I am not really sure about the types here, they tend to get a bit too complicated maybe... `FormData<>` will limit the keys of the object to only those that are strings. That was needed, because we map the keys to what users specify as their fields `@name`, an the latter can only be strings. But keys of type `object` can besides strings also be `number` or `symbol`. So `FormData` is getting rid of those. 

But `validate` will actually receive the whole `@data` object, including _potentially_ keys of type `number | symbol`, and TS was complaining about mismatching types for the `validateChangset` helper due to this...

Again, not super happy about all of this, as this does not really matter in the real world, it is more to just make TS shut up complaining about subtle type errors 🙄

Reply:
what is correct though? We can coerce TS in to doing our bidding (I have lots of experience with this with getting ember-resources and ember-statechart-component glint-compatible 😅 )